lib/pbench/test/unit/server/database/__init__.py

```diff
@@ -76,40 +72,87 @@ class FakeRow:
     """
 
     def __init__(self, cls: Optional[type[Database.Base]] = None, **kwargs):
+        """Construct a FakeRow object from a set of attributes
+
+        Args:
+            cls:    A SQLAlchemy database object class
+            kwargs: DB row column key/value pairs
+        """
         self.cls = cls
         for k, v in kwargs.items():
             setattr(self, k, v)
 
     def _columns(self) -> list[str]:
+        """Internal helper to filter out the column declarations from either a
+        real SQLAlchemy proxy object or the cloned attributes from a FakeRow
+        """
         return [c.name for c in self.cls.__table__._columns] if self.cls else dir(self)
 
     @classmethod
     def clone(cls, template: Database.Base) -> "FakeRow":
+        """Factory to create a FakeRow from a 'template' SQLAlchemy proxy
+        object through introspection.
+
+        Args:
+            template:   A SQLAlchemy Database object subclass
+
+        Returns:
+            A FakeRow object with attributes representing the SQLAlchemy proxy
+            object's columns.
+        """
         new = cls(cls=type(template))
         for c in new._columns():
             setattr(new, c, getattr(template, c))
         return new
 
     def __repr__(self) -> str:
+        """Return a string representing the row"""
         return (
             "Row(" + ",".join(f"{c}={getattr(self, c)}" for c in self._columns()) + ")"
         )
 
     def __eq__(self, entity) -> bool:
+        """Compare this object with another for equality. The other object must
+        be either a FakeRow or a Database proxy class matching the class
+        association of the fake.
```

Comment:
I suggest reworking the second sentence a bit.  The requirement that the "other object" be another instance of the same class is pretty normal for an `__eq__()` implementation, so it hardly bears mentioning; rather, the fact that it could instead be the class object is the key thing that we should convey.  How about,

> The other object may be either an instance of the same class or a reference to the class object itself (i.e., the Database proxy class).